src/main/adapters/route-adapter.ts

```diff
@@ -2,7 +2,7 @@ import { Request, Response } from 'express'
 import { Controller } from '../../presentation/protocols/controller'
 
 export default (controller: Controller) => {
-  return async (req: Request, res: Response) => {
+  return async (req: Request, res: Response): Promise<any> => {
```

Comment:
Why any type?

Reply:
TypeScript is not recognizing  (req: Request, res: Response) and (req: Request, res: Response, next: NextFunction) as a correct type but it is